sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

```diff
@@ -40,6 +57,59 @@
     this.batchClient = batchClient;
   }
 
+  static SpannerAccessor create(SpannerConfig spannerConfig) {
+    SpannerOptions.Builder builder = SpannerOptions.newBuilder();
+
+    ValueProvider<Duration> commitDeadline = spannerConfig.getCommitDeadline();
+    if (commitDeadline != null && commitDeadline.get().getMillis() > 0) {
+
+      // In Spanner API version 1.21 or above, we can set the deadline / total Timeout on an API
+      // call using the following code:
+      //
+      // UnaryCallSettings.Builder commitSettings =
+      // builder.getSpannerStubSettingsBuilder().commitSettings();
+      // RetrySettings.Builder commitRetrySettings = commitSettings.getRetrySettings().toBuilder()
+      // commitSettings.setRetrySettings(
+      //     commitRetrySettings.setTotalTimeout(
+      //         Duration.ofMillis(getCommitDeadlineMillis().get()))
+      //     .build());
+      //
+      // However, at time of this commit, the Spanner API is at only at v1.6.0, where the only
+      // method to set a deadline is with GRPC Interceptors, so we have to use that...
+      SpannerInterceptorProvider interceptorProvider =
+          SpannerInterceptorProvider.createDefault()
+              .with(new CommitDeadlineSettingInterceptor(commitDeadline.get()));
+      builder.setInterceptorProvider(interceptorProvider);
+    }
+
+    ValueProvider<String> projectId = spannerConfig.getProjectId();
+    if (projectId != null) {
+      builder.setProjectId(projectId.get());
+    }
+    ServiceFactory<Spanner, SpannerOptions> serviceFactory = spannerConfig.getServiceFactory();
+    if (serviceFactory != null) {
+      builder.setServiceFactory(serviceFactory);
+    }
+    ValueProvider<String> host = spannerConfig.getHost();
+    if (host != null) {
+      builder.setHost(host.get());
+    }
+    String userAgentString = USER_AGENT_PREFIX + "/" + ReleaseInfo.getReleaseInfo().getVersion();
+    builder.setHeaderProvider(FixedHeaderProvider.create("user-agent", userAgentString));
+    SpannerOptions options = builder.build();
+
+    Spanner spanner = options.getService();
+    String instanceId = spannerConfig.getInstanceId().get();
+    String databaseId = spannerConfig.getDatabaseId().get();
+    DatabaseClient databaseClient =
+        spanner.getDatabaseClient(DatabaseId.of(options.getProjectId(), instanceId, databaseId));
+    BatchClient batchClient =
+        spanner.getBatchClient(DatabaseId.of(options.getProjectId(), instanceId, databaseId));
+    DatabaseAdminClient databaseAdminClient = spanner.getDatabaseAdminClient();
+
+    return new SpannerAccessor(spanner, databaseClient, databaseAdminClient, batchClient);
+  }
+
   public DatabaseClient getDatabaseClient() {
```

Comment:
I think this line is the cause of the API surface leak, because all the test failure error messages have 

```
...
		interface com.google.cloud.spanner.DatabaseAdminClient
		class org.apache.beam.sdk.io.gcp.spanner.SpannerAccessor
```

Reply:
The commit message has 

```
Expose DatabaseAdminClient

Change-Id: I50670b8cd6a479fcad6ff866c6adb97434ae2ed8
```